pkgs/servers/calibre-web/default.nix

```diff
@@ -0,0 +1,67 @@
+{ lib
+, fetchFromGitHub
+, python3
+, python3Packages
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "calibre-web";
+  version = "0.6.11";
+  src = fetchFromGitHub {
+    owner = "janeczku";
+    repo = "calibre-web";
+    rev = version;
+    sha256 = "10sjllhhcamswpa1wlim4mbm2zl4g804bwly5p4nmklg7n1v226g";
+  };
+
+  prePatch = ''
+    substituteInPlace setup.cfg \
+        --replace "requests>=2.11.1,<2.25.0" "requests>=2.11.1,<=2.25.1" \
+        --replace "cps = calibreweb:main" "calibre-web = calibreweb:main"
+  '';
+
+  patches = [
+    # default-logger.patch switches default logger to /dev/stdout. Otherwise calibre-web tries to open a file relative
+    # to its location, which can't be done as the store is read-only. Log file location can later be configured using UI
+    # if needed.
+    ./default-logger.patch
+    # DB migrations adds an '-m' CLI option that instructs calibre-web to run DB migrations and exit. This is gonna be
+    # used to configure calibre-web declaratively, as most of its configuration parameters are stored in the DB.
+    ./db-migrations.patch
```

Comment:
I don't think packaging should add new options. Is there an upstream commit with those changes we could fetch?

Reply:
> I don't think packaging should add new options.

I agree. I was rather reluctant to add it. But the only way to configure `calibre-web` is through the sqlite DB. So if we want to have a declarative module, and I'd like to have one, we need to have a way to ensure that DB exists even before the first start.

The only option that can be set is the listening port (via env var). This variable is only consulted on the first start when there is no DB. So even that can't be put into module.

There is an upstream issue (https://github.com/janeczku/calibre-web/issues/1856) with a proposal to add configuration file support. But before it gets implemented (if it gets implemented at all), there is basically no way to create a module.

If you really don't like the patch, I can drop it and settle with configuring `calibre-web` via UI.

> Is there an upstream commit with those changes we could fetch?

No.